lua/autorun/server/sv_restrict_propcore.lua

```diff
@@ -22,22 +22,42 @@ function restrictPropCoreFunctions()
         --"propBreak(e:)"
     }
 
-    for _, signature in pairs( restrictedFunctions ) do
-        if wire_expression2_funcs then
-            local oldFunc = wire_expression2_funcs[signature][3]
-
-            wire_expression2_funcs[signature][3] = function( self, ... )
-                if ( disallowedRanks[self.player:GetUserGroup()] == nil ) then
-                    local isInBuildMode = self.player:GetNWBool("CFC_PvP_Mode", false) == false
-
-                    if( isInBuildMode or self.player:IsAdmin() ) then
-                        return oldFunc( self, ... )
-                    else
-                        self.player:ChatPrint( "You can't use PropCore in PvP mode" )
-                    end
-                else
-                    self.player:ChatPrint( "You don't have access to " .. signature )
-                end
+    local adminOnlyFunctions = {
+        "use(e:)"
+    }
+
+    restrict( restrictedFunctions, function( self,  ... )
+        if disallowedRanks[self.player:GetUserGroup()] then
+            return false, "You don't have access to this function"
+        elseif self.player:GetNWBool( "CFC_PvP_Mode" ) == true then
```

Comment:
you might want to set  it to a local variable like was done originally
`local isInBuildMode = self.player:GetNWBool("CFC_PvP_Mode", false) == false`
also if they are an admin the pvp mode restrictions should be  ignored